basxconnect/mailer_integration/download_data.py

```diff
@@ -0,0 +1,93 @@
+from typing import Any, NamedTuple
+
+import basxconnect
+from basxconnect.core import models
+from basxconnect.mailer_integration.abstract.abstract_datasource import Datasource
+from basxconnect.mailer_integration.models import Interest, MailingPreferences
+
+
+class SynchronizationResult(NamedTuple):
+    new_persons: int
+    total_synchronized_persons: int
+
+
+def download_persons(datasource: Datasource) -> SynchronizationResult:
+    MailingPreferences.objects.all().delete()
+    Interest.objects.all().delete()
```

Comment:
This is a bigger issue with synchronization: If we (meaning basxconnect) cannot guarantee that basxconnect users never modify these values and if we cannot guarantee that the download always succeeds then we are guaranteed to have an inconsistent state at some point. The second issue could maybe resolved by wrapping the whole download-functionality inside a transaction and make sure the transaction gets rolled-back in case of a failure (see https://docs.djangoproject.com/en/dev/topics/db/transactions/#controlling-transactions-explicitly)

Reply:
I think we should discuss the synchronization issue a bit more in detail anyway. But let us wait until the next meeting, my understanding was, that Beat wanted to say something about synchronization too.

Reply:
If the downloading fails we can just retry it and then the previous failure won't matter any more. It should not leave any traces.